nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Utilities/SdkPackageUpdaterTests.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace NuGetUpdater.Core.Test.Utilities;
+
+public class SdkPackageUpdaterTests
+{
+    [Theory]
+    [MemberData(nameof(GetDependencyUpdates))]
+    public async Task UpdateDependency_UpdatesDependencies((string Path, string Contents)[] startingContents, (string Path, string Contents)[] expectedContents,
+        string dependencyName, string previousVersion, string newDependencyVersion, bool isTransitive)
+    {
+        // Arrange
+        using var directory = TemporaryDirectory.CreateWithContents(startingContents);
+        var projectPath = Path.Combine(directory.DirectoryPath, startingContents.First().Path);
+        var logger = new Logger(verbose: false);
+        // Act
+        await SdkPackageUpdater.UpdateDependencyAsync(directory.DirectoryPath, projectPath, dependencyName, previousVersion, newDependencyVersion, isTransitive, logger);
+
+        // Assert
+        AssertContentsEqual(expectedContents, directory);
+    }
+
+    public static IEnumerable<object[]> GetDependencyUpdates()
+    {
+        // Simple case
+        yield return new object[]
```

Comment:
I tried to make it simple to add additional cases we want to verify here but also didn't want to bog down the already complicated PR with a ton of examples.

Reply:
Personally, I don't mind extra stuff to review in a PR if it improves test coverage.